lib/jellyfish/resource_manager.ex

```diff
@@ -0,0 +1,79 @@
+defmodule Jellyfish.ResourceManager do
+  @moduledoc """
+  Module responsible for deleting outdated resources.
+  Right now it only removes outdated resources created by recording component.
+  """
+
+  use GenServer, restart: :temporary
+
+  require Logger
+
+  alias Jellyfish.Component.Recording
+
+  @type seconds :: pos_integer()
+  @type opts :: %{interval: seconds(), recording_timeout: seconds()}
+
+  @spec start(opts()) :: {:ok, pid()} | {:error, term()}
+  def start(opts), do: GenServer.start(__MODULE__, opts)
+
+  @spec start_link(opts()) :: GenServer.on_start()
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  @impl true
+  def init(opts) do
+    Logger.debug("Initialize resource manager")
+
+    send(self(), :free_resources)
+
+    {:ok, opts}
+  end
+
+  @impl true
+  def handle_info(:free_resources, state) do
+    base_path = Recording.get_base_path()
+
+    with {:ok, files} <- File.ls(base_path) do
+      current_time = System.system_time(:second)
+
+      files
+      |> Enum.map(&Path.join(base_path, &1))
+      |> Enum.each(&remove_recording_if_obsolete(current_time, state.recording_timeout, &1))
+    else
+      {:error, reason} ->
+        Logger.error("Resource Manager: can't list recordings, reason: #{reason}")
+    end
+
+    Process.send_after(self(), :free_resources, state.interval * 1000)
```

Comment:
suggestion: I think we could adjust accordingly to `GenServer` tutorial https://hexdocs.pm/elixir/1.16.2/GenServer.html#module-receiving-regular-messages